Source/ToolCore/JSBind/JSBTypeScript.cpp

```diff
@@ -57,6 +65,9 @@ String JSBTypeScript::GetScriptType(JSBFunctionType* ftype)
         scriptType = "string[]";
     }
 
+    if (ftype->type_->asHeapPtrType())
+        scriptType = "any";
+
```

Comment:
I was with one of the callback registering functions on components I think. Since the callback parameter wasn't a recognized type it resolved to a Number, which TypeScript wouldn't accept. First prize would of course be to have the callback types be properly resolved to proper TS callback syntax, but that was a much bigger add.